app/components/book-appointment.js

```diff
@@ -1,28 +1,10 @@
 import Ember from "ember";
 
 export default Ember.Component.extend({
-  isDetailsComplete() {
-    const user = this.get("session.currentUser");
-    if (!user) {
-      return false;
-    }
-
-    const organisationsUser = user.get("organisationsUsers.firstObject");
-    const organisation =
-      organisationsUser && organisationsUser.get("organisation");
-    const hasInfoAndCharityRole =
-      user.get("isInfoComplete") && user.hasRole("Charity");
-    const hasCompleteOrganisationUserInfo =
-      organisationsUser && organisationsUser.get("isInfoComplete");
-
-    return (
-      hasInfoAndCharityRole && organisation && hasCompleteOrganisationUserInfo
-    );
-  },
-
+  session: Ember.inject.service(),
   actions: {
     redirectAsPerUserDetails() {
-      if (this.isDetailsComplete()) {
+      if (this.get("session").accountDetailsComplete()) {
```

Comment:
Great! Abstract behind a service and re-use elsewhere

Reply:
👍